nexus/tests/integration_tests/subnet_allocation.rs

```diff
@@ -72,53 +84,91 @@ async fn test_subnet_allocation(cptestctx: &ControlPlaneTestContext) {
         organization_name, project_name
     );
 
-    // Modify the default VPC to have a very small subnet so we don't need to
-    // issue many requests
-    let url_subnet = format!(
-        "/organizations/{}/projects/{}/vpcs/default/subnets/default",
+    // Create a new, small VPC Subnet, so we don't need to issue many requests
+    // to test address exhaustion.
+    let url_subnets = format!(
+        "/organizations/{}/projects/{}/vpcs/default/subnets",
         organization_name, project_name
     );
-    let subnet = "192.168.42.0/29".parse().unwrap();
-    let subnet_update = params::VpcSubnetUpdate {
-        identity: IdentityMetadataUpdateParams {
-            name: Some("default".parse().unwrap()),
-            description: None,
+    let subnet_name = "small";
+    let subnet = "192.168.42.0/26".parse().unwrap();
+    let subnet_create = params::VpcSubnetCreate {
+        identity: IdentityMetadataCreateParams {
+            name: subnet_name.parse().unwrap(),
+            description: String::from("a small subnet"),
         },
-        ipv4_block: Some(Ipv4Net(subnet)),
+        // Use the minimum subnet size
+        ipv4_block: Ipv4Net(subnet),
         ipv6_block: None,
     };
-    NexusRequest::object_put(client, &url_subnet, Some(&subnet_update))
+    NexusRequest::objects_post(client, &url_subnets, &Some(&subnet_create))
         .authn_as(AuthnMode::PrivilegedUser)
         .execute()
         .await
         .unwrap();
 
     // The valid addresses for allocation in `subnet` are 192.168.42.5 and
     // 192.168.42.6. The rest are reserved as described in RFD21.
-    create_instance(client, organization_name, project_name, "i1").await;
-    create_instance(client, organization_name, project_name, "i2").await;
+    let nic = params::InstanceNetworkInterfaceAttachment::Create(vec![
+        params::NetworkInterfaceCreate {
+            identity: IdentityMetadataCreateParams {
+                name: "eth0".parse().unwrap(),
+                description: String::from("some iface"),
+            },
+            vpc_name: "default".parse().unwrap(),
+            subnet_name: "small".parse().unwrap(),
+            ip: None,
+        },
+    ]);
+
+    // Create a shit-ton of instances, up to the size of the IP subnet.
+    //
+    // The first 5 addresses in an IP subnet are reserved, as is the broadcast
```

Comment:
This feels like a constant that we should be able share between implementation / tests

Reply:
It is a constant in `nexus::defaults`, but that module isn't public. I can make it so?

Reply:
I see there are other such usually-private modules that we've made public for testing. I'll do that with the defaults.

Reply:
Yeah, we can either re-export the module, or just the constants we need. Since nexus is basically a binary, not a library, we have leeway.